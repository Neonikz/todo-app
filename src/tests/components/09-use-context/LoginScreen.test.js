import React from 'react'
import { LoginScreen } from '../../../components/09-use-context/LoginScreen';
import { UserContext } from '../../../components/09-use-context/UserContext';
import {mount} from 'enzyme';

describe('Pruebas en <LoginScreen />', () => {
    
    const user = {
        id: 123,
        name: 'Nicolas'
    }
    const setUser = jest.fn();

    const wrapper = mount( 
        <UserContext.Provider value={{
            setUser
        }}>
            <LoginScreen />
        </UserContext.Provider>
    );

    test('Debe de mostrar todo correctamente', () => {
        expect(wrapper).toMatchSnapshot();
    });
    
    test('Debe de ejecutar el setUser con el argumento esperado', () => {
        wrapper.find('button').prop('onClick')(user);
        expect(setUser).toHaveBeenCalledWith( {
            id: 123,
            name: 'Nicolas'
        } );
    });
    
});
