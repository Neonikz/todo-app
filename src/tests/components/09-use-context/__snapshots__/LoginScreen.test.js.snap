// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <LoginScreen /> Debe de mostrar todo correctamente 1`] = `
Array [
  <h1>
    Login Screen
  </h1>,
  <hr />,
  <button
    className="btn btn-outline-primary"
    onClick={[Function]}
  >
    Login
  </button>,
]
`;
